xds/src/test/java/io/grpc/xds/LocalityStoreTest.java

```diff
@@ -266,6 +269,68 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
+    Map<Locality, LocalityInfo> localityInfoMap = ImmutableMap.of(
+        locality1, localityInfo1, locality2, localityInfo2, locality3, localityInfo3);
+    localityStore.updateLocalityStore(localityInfoMap);
+    localityStore.updateDropPercentage(ImmutableList.of(
+        new DropOverload("throttle", 365),
+        new DropOverload("lb", 1234)));
+
+    assertThat(loadBalancers).hasSize(3);
+    ArgumentCaptor<ResolvedAddresses> resolvedAddressesCaptor1 =
+        ArgumentCaptor.forClass(ResolvedAddresses.class);
+    verify(loadBalancers.get(0)).handleResolvedAddresses(resolvedAddressesCaptor1.capture());
+    assertThat(resolvedAddressesCaptor1.getValue().getAddresses()).containsExactly(eag11, eag12);
+    ArgumentCaptor<ResolvedAddresses> resolvedAddressesCaptor2 =
+        ArgumentCaptor.forClass(ResolvedAddresses.class);
+    verify(loadBalancers.get(1)).handleResolvedAddresses(resolvedAddressesCaptor2.capture());
+    assertThat(resolvedAddressesCaptor2.getValue().getAddresses()).containsExactly(eag21, eag22);
+    ArgumentCaptor<ResolvedAddresses> resolvedAddressesCaptor3 =
+        ArgumentCaptor.forClass(ResolvedAddresses.class);
+    verify(loadBalancers.get(2)).handleResolvedAddresses(resolvedAddressesCaptor3.capture());
+    assertThat(resolvedAddressesCaptor3.getValue().getAddresses()).containsExactly(eag31, eag32);
+    assertThat(pickerFactory.totalReadyLocalities).isEqualTo(0);
+
+    // subchannel12 goes to CONNECTING
+    final Subchannel subchannel12 =
+        helpers.get(0).createSubchannel(ImmutableList.of(eag12), Attributes.EMPTY);
+    verify(helper).createSubchannel(ImmutableList.of(eag12), Attributes.EMPTY);
+    SubchannelPicker subchannelPicker12 = new SubchannelPicker() {
+      @Override
+      public PickResult pickSubchannel(PickSubchannelArgs args) {
+        return PickResult.withSubchannel(subchannel12);
+      }
+    };
+    helpers.get(0).updateBalancingState(CONNECTING, subchannelPicker12);
+    ArgumentCaptor<SubchannelPicker> subchannelPickerCaptor12 =
+        ArgumentCaptor.forClass(SubchannelPicker.class);
+    verify(helper).updateBalancingState(same(CONNECTING), subchannelPickerCaptor12.capture());
+
+    doReturn(365, 1234).when(random).nextInt(1000_000);
+    assertThat(subchannelPickerCaptor12.getValue().pickSubchannel(pickSubchannelArgs))
+        .isEqualTo(PickResult.withNoResult());
+
+    doReturn(366, 1235).when(random).nextInt(1000_000);
+    assertThat(subchannelPickerCaptor12.getValue().pickSubchannel(pickSubchannelArgs))
+        .isEqualTo(PickResult.withNoResult());
+
+    doReturn(364, 1234).when(random).nextInt(1000_000);
+    assertThat(subchannelPickerCaptor12.getValue().pickSubchannel(pickSubchannelArgs).isDrop())
+        .isTrue();
+
+    doReturn(365, 1233).when(random).nextInt(1000_000);
+    assertThat(subchannelPickerCaptor12.getValue().pickSubchannel(pickSubchannelArgs).isDrop())
+        .isTrue();
+  }
```

Comment:
I would also verify the times random.nextInt() is called after each pick.

Reply:
Done.